### PR TITLE
Feature/models tweaking and testing

### PR DESCRIPTION
**Pull request by CrashingBrain**
Please approve and take vision of the new Models <3

### CODE REVIEW

File: models/Freelance.js [hunk 1]
```diff
@@ -19,15 +19,21 @@ const ObjectID = mongoose.Schema.Types.ObjectId;
...
-		price			: { type: Object, required: true },
-		avgScore 		: { type: Number },
-		reviews			: [{ type: ObjectID, ref: "Review", default: [] }],
-		tags			: [{ type: String, default: [] }],
+		firstName			: { type: String, required: true },
+		familyName		: { type: String },
+		title					: { type: String },
+		category			: { type: String },
+		description		: { type: String },
+		urlPic				: { type: String },
```
Comment (BalestraPatrick) [inline models/Freelance.js]: Would you mind naming this property `urlPicture`?

File: models/Freelance.js [hunk 2]
```diff
@@ -38,24 +44,31 @@ const Freelance = exports.Freelance = new mongoose.Schema({
 Freelance.pre('save', function (next) {
 
 	//review between 0 and 5
-	if (this.avgScore > 5) {
-		this.avgScore = 5;
-	} else if (this.avgScore < 0) {
-		this.avgScore = 0;
-		//maybe a problem with default -1 value of review, to check later.
+	if ( this.avgScore !== undefined ){
```
Comment (BalestraPatrick) [inline models/Freelance.js]: Could you please format the code in the usual way like it was before? `if (this.avgScore !== undefined) {`

**Review by BalestraPatrick — CHANGES_REQUESTED** [review]

File: models/Freelance.js [hunk 3]
```diff
@@ -38,24 +44,31 @@ const Freelance = exports.Freelance = new mongoose.Schema({
...
 	//and that they are both above 0; in particular, max must be > min
-	if (!(this.price.hasOwnProperty('min') && this.price.hasOwnProperty('max'))) {
-		this.price = {min: 0, max: 0};
-	}
-	if (this.price.min < 0) {
-		this.price.min = 0;
-	} else if (this.price.max < 0 || this.price.max < this.price.min) {
-		this.price.max = this.price.min;
+	//maybe in future
+	if ( this.price !== undefined ){
```
Comment (larabr) [inline models/Freelance.js]: space after `)` and not inside `(` and `)` thanks

File: models/Freelance.js [hunk 4]
```diff
@@ -38,24 +44,31 @@ const Freelance = exports.Freelance = new mongoose.Schema({
...
+		if ((this.price.hasOwnProperty('min') && this.price.hasOwnProperty('max'))) {
+			if (this.price.min < 0) {
+				this.price.min = 0;
+			} else if (this.price.max < 0 || this.price.max < this.price.min) {
+				this.price.max = this.price.min;
+			}
+		} else{
```
Comment (larabr) [inline models/Freelance.js]: space after `else` thanks

File: models/Review.js [hunk 5]
```diff
@@ -32,6 +34,15 @@ Review.pre('save', function (next) {
 	} else if (this.score < 0) {
 		this.score = 0;
 	}
+
+	if (this.text !== undefined){
+		let length = this.text.length;
+
+		if(length>1000){
```
Comment (larabr) [inline models/Review.js]: spaces around `>` and before `(` and after `)` thanks

File: models/Tag.js [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+	//in future maybe add User and author is ObjectID ref User
+		tagName 		: { type: String, required: true },
+		freelancers 	: [{ type: ObjectID, ref: 'Freelance', default:[] }],
+	},
+	{
+		versionKey: false,
+	}
+);
+
+Tag.pre('save', function (next) {
```
Comment (larabr) [inline models/Tag.js]: no space after `function` thanks

File: models/Freelance.js [hunk 7]
```diff
@@ -19,15 +19,21 @@ const ObjectID = mongoose.Schema.Types.ObjectId;
...
-		address			: { type: String },
-		email			: { type: String, required: true },
-		phone			: { type: String },
-		price			: { type: Object, required: true },
-		avgScore 		: { type: Number },
-		reviews			: [{ type: ObjectID, ref: "Review", default: [] }],
-		tags			: [{ type: String, default: [] }],
+		firstName			: { type: String, required: true },
+		familyName		: { type: String },
+		title					: { type: String },
```
Comment (aronfiechter) [inline models/Freelance.js]: Can we please make `title` a required property? As said, it's as important as the `firstName`.
A practical reason is: a person that wants to make two profiles for two different jobs, maybe both in the same category.

File: models/Freelance.js [hunk 8]
```diff
@@ -19,15 +19,21 @@ const ObjectID = mongoose.Schema.Types.ObjectId;
...
+		description		: { type: String },
+		urlPic				: { type: String },
+		address				: { type: String },
+		email					: { type: String, required: true },
+		phone					: { type: String },
+		price					: { type: Object },
+		//we recompute this on every review
+		avgScore 			: { type: Number },
+		reviews				: [{ type: ObjectID, ref: "Review", default: [] }],
+		tags					: [{ type: ObjectID, default: [] }],
```
Comment (aronfiechter) [inline models/Freelance.js]: We should add `ref: "Tag", ` similarly to how `reviews` is.

File: seed_data/seedData.js [hunk 9]
```diff
@@ -12,37 +12,49 @@ var freelancers = {
       "_id"         : "5837133bcb98f316ac473853",
       "firstName"   : "John",
       "familyName"  : "Smith",
+      "title"       : "President Candidate",
```
Comment (aronfiechter) [inline seed_data/seedData.js:15]: Not really a "job" that a freelancer can offer. What would you hire a `President Candidate` for (except making America great again)?

File: seed_data/seedData.js [hunk 10]
```diff
@@ -12,37 +12,49 @@ var freelancers = {
...
       "price"       : { min : 50, max : 100 },
       "reviews"     : ["5866a5cb2d6c4804f847ea37", "5866a5cb2d6c4804f847ea36"],
-      "tags"        : ["testTag1", "testTag3", "testTag4"]
+      "tags"        : ["585eaac6f1dc05033ee9ce97"]
     },
     {
       "_id"         : "5837133bcb98f316ac473852",
       "firstName"   : "J.R.R.",
       "familyName"  : "Tolkien",
+      "title"       : "Blogger",
```
Comment (aronfiechter) [inline seed_data/seedData.js:47]: Is "Blogger" a job you can pay someone to do for you?

**Review by aronfiechter — CHANGES_REQUESTED** [review]
Please really let's make Title a required field, so that a basic Freelancer Profile has:

**[Job Title - Name]**
*email*

File: seed_data/seedData.js (hunk 9 above)
Comment (larabr) [reply seed_data/seedData.js:15]: Isn't it `Presidential Candidate` anyway?

File: seed_data/seedData.js (hunk 10 above)
Comment (larabr) [reply seed_data/seedData.js:47]: Yes, int the content writer sense - anyway that's up to the user so
Why are categories hard coded - unlike tags?

File: models/Freelance.js (hunk 1 above)
Comment (CrashingBrain) [reply models/Freelance.js]: done

File: models/Freelance.js (hunk 2 above)
Comment (CrashingBrain) [reply models/Freelance.js]: done

File: models/Freelance.js (hunk 3 above)
Comment (CrashingBrain) [reply models/Freelance.js]: done

File: models/Freelance.js (hunk 4 above)
Comment (CrashingBrain) [reply models/Freelance.js]: done

File: models/Review.js (hunk 5 above)
Comment (CrashingBrain) [reply models/Review.js]: done

File: models/Tag.js (hunk 6 above)
Comment (CrashingBrain) [reply models/Tag.js]: done

File: models/Freelance.js (hunk 7 above)
Comment (CrashingBrain) [reply models/Freelance.js]: ok. done

File: models/Freelance.js (hunk 8 above)
Comment (CrashingBrain) [reply models/Freelance.js]: yep. right. done.

**Review by aronfiechter — APPROVED** [review]

**Pull request merged**